multi-cluster-orchestrator/samples/hello_world/1-infrastructure/main.tf

```diff
@@ -0,0 +1,298 @@
+  display_name = "Service Account for ${each.key} cluster"
+}
+
+# Cluster Service Account Permissions
+resource "google_project_iam_member" "clusters" {
+  for_each = {
+    for o in distinct(flatten([
+      for sa in google_service_account.clusters :
+      [
+        for role in [
+          "roles/container.defaultNodeServiceAccount",
+          "roles/monitoring.metricWriter",
+          # For image streaming
+          "roles/serviceusage.serviceUsageConsumer"
+        ] :
+        {
+          "email" : sa.email,
+          "role" : role,
+        }
+      ]
+    ])) :
+    "${o.email}/${o.role}" => o
+  }
+
+  project = data.google_project.default.project_id
+  role    = each.value.role
+  member  = "serviceAccount:${each.value.email}"
+}
+
+# Workload Identity Permissions
+
+# Service Account of the ArgoCD Application Controller
+# Syncs the applications directly to the workload clusters
+resource "google_project_iam_member" "hub-wi-argo-appsync" {
+  project = data.google_project.default.project_id
+  for_each = toset([
+    "roles/gkehub.viewer",
+    "roles/container.developer",
+    "roles/gkehub.gatewayEditor"
+  ])
+
+  role   = each.value
+  member = "principal://iam.googleapis.com/projects/${data.google_project.default.number}/locations/global/workloadIdentityPools/${data.google_project.default.project_id}.svc.id.goog/subject/ns/argocd/sa/argocd-application-controller"
+
+  # Requires Identity Pool
+  depends_on = [google_container_cluster.hub]
+}
+
+# Service Account of the Orchestrator Controller Manager
+resource "google_project_iam_member" "hub-wi-mco-monitoring" {
+  project = data.google_project.default.project_id
+
+  role   = "roles/monitoring.viewer"
+  member = "principal://iam.googleapis.com/projects/${data.google_project.default.number}/locations/global/workloadIdentityPools/${data.google_project.default.project_id}.svc.id.goog/subject/ns/orchestra-system/sa/orchestra-controller-manager"
+
+  # Requires Identity Pool
+  depends_on = [google_container_cluster.hub]
+}
+
+# Service Account of the Custom Metrics Stackdriver Adapter
+resource "google_project_iam_member" "hub-wi-custom-metrics" {
+  project = data.google_project.default.project_id
+
+  role   = "roles/monitoring.viewer"
+  member = "principal://iam.googleapis.com/projects/${data.google_project.default.number}/locations/global/workloadIdentityPools/${data.google_project.default.project_id}.svc.id.goog/subject/ns/custom-metrics/sa/custom-metrics-stackdriver-adapter"
+
+  # Requires Identity Pool
+  depends_on = [google_container_cluster.hub]
+}
+
+# Service Account of the MCS Importer
+# https://cloud.google.com/kubernetes-engine/docs/how-to/multi-cluster-services#enabling
+resource "google_project_iam_member" "gke-mcs-importer" {
+  project = data.google_project.default.project_id
+
+  role   = "roles/compute.networkViewer"
+  member = "serviceAccount:${data.google_project.default.project_id}.svc.id.goog[gke-mcs/gke-mcs-importer]"
```

Comment:
this syntax makes me sad; we should check with them if they can support the new principal:// syntax. (they likely can unless they are GKE RBAC stuff)

Reply:
It's not documented, but I'll test it as it looks like it "should" work

Reply:
Everything looks good with the new principal syntax version.